JsonPreprocessor/CJsonPreprocessor.py

```diff
@@ -590,7 +590,7 @@ def __checkAndUpdateKeyValue(self, sInputStr: str) -> str:
 
       Nested param "${abc}['xyz']" -> "str(${abc}['xyz'])"
         '''
-        if re.search("\"\s*[0-9A-Za-z\.\-_\s]*\s*\${\s*[0-9A-Za-z\.\-_]+\s*}(\[+\s*.+\s*\]+)*\s*[0-9A-Za-z\.\-_\s]*\s*\"", sInputStr.lower()):
+        if re.search("\"\s*[0-9A-Za-z@#%&*+=\\:,;,?<>~().\-_\s]*\s*\${\s*[0-9A-Za-z\.\-_]+\s*}(\[+\s*.+\s*\]+)*\s*[0-9A-Za-z@#%&*+=\\:,;,?<>~().\-_\s]*\s*\"", sInputStr.lower()):
```

Comment:
Hi Son,
you pattern allows . (dot) and - (hyphen) as start of a variable name.
This is not correct.
I think  `"\${\s*[0-9A-Za-z_]+[0-9A-Za-z\.\-_]*\s*}"` should do better. Similar I do in TML.
Thank you,
 Thomas

Reply:
Hello Thomas,

Actually, the patten **"\"\s*[0-9A-Za-z@#%&*+=\\:,;,?<>~().\-_\s]*\s*** is not a part of variable name, it is a part of the string value which contains nested variable inside. For example, the string value of parameter could be "My email is namsonx@gmail.com ${nested_param}, and its status is active", and the value of parameter "nested_param" will be replaced in this string value.

Thank you,
Son

Reply:
Hi Son,
1. `"My email is [namsonx@gmail.com](mailto:namsonx@gmail.com) ${..--nested_param}, and its status is active"` matches to you pattern, but is wrong. No dot or hyphen allowed at beginning.
2. Did you try `"Email của tôi là [namsonx@gmail.com](mailto:namsonx@gmail.com) ${nested_param}, và trạng thái của nó là đang hoạt động"`
Thank you,
 Thomas

Reply:
Hello Thomas,

1. I will update the pattern ${\s*[0-9A-Za-z\.\-_]+\s*} to avoid dot or hyphen is not at the beginning of variable name.
2. I have just tested, and utf-8 characters do not work :) , I will update the change tomorrow.
Thank you for your comments.

Thank you,
Son

Reply:
very good!